### PR TITLE
TBRANDS-28 - Story carousel link component

### PR DESCRIPTION
**Pull request by matthewroach**
## Description

Update Story Carousel Block to use the `Link` component

## Jira Ticket

- [TBRANDS-28](https://arcpublishing.atlassian.net/browse/TBRANDS-28)

## Test Steps

- Add test steps a reviewer must complete to test this PR

1. Checkout this branch `git checkout story-carousel-link-component`
2. Using the `commerce` branch - run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/story-carousel-block`
3. Add story block to a page and configure custom fields
4. Validate no changes in appearance, and links still work as before

## Author Checklist

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x] Confirmed this PR has unit test files
  - [x] Ran `npm run test`, made sure all tests are passing
  - [x] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

**Review by JackHowa — APPROVED** [review]
Good work

**Comment by brentmiller1973** [comment]
@matthewroach I noticed there is an a11y violation with the Link being used:

image

When I revert back to using a vanilla `a` element, the violation is not there:

image

**Review by brentmiller1973 — COMMENTED** [review]
@matthewroach See my comment about the a11y violation on the Link component

File: blocks/story-carousel-block/features/story-carousel/default.jsx [hunk 1]
```diff
@@ -115,7 +116,7 @@ const StoryCarousel = ({
 									maximum: elements.length,
 								})}`}
 							>
-								<a className={`${BLOCK_CLASS_NAME}__story-card`} href={url}>
+								<Link className={`${BLOCK_CLASS_NAME}__story-card`} href={url}>
```
Comment (brentmiller1973) [inline blocks/story-carousel-block/features/story-carousel/default.jsx]: This removes the a11y violation

`<Link tabindex='-1' className={`${BLOCK_CLASS_NAME}__story-card`} href={url}>`

**Review by brentmiller1973 — CHANGES_REQUESTED** [review]

File: blocks/story-carousel-block/features/story-carousel/default.jsx (hunk 1 above)
Comment (matthewroach) [reply blocks/story-carousel-block/features/story-carousel/default.jsx]: Updated to use the `assistiveHidden` prop

Comment (brentmiller1973) [reply blocks/story-carousel-block/features/story-carousel/default.jsx]: awesome!

**Review by brentmiller1973 — APPROVED** [review]

**Pull request merged**